scripting/include/srccoop/deathnotice.inc

```diff
@@ -0,0 +1,547 @@
+		DnManager.GetDisplayNameForEvent(pAttacker, sEntry.m_szTarget1Name, sizeof(sEntry.m_szTarget1Name));
+
+		if (pVictim.IsClassPlayer())
+		{
+			// Check if feature is enabled.
+			if (g_sDnData.m_piNoticeEntityKills.IntValue == 0)
+				return false;
+
+			// The victim is a player so it can be used directly.
+			sEntry.m_pTarget2 = view_as<CBasePlayer>(pVictim);
+			sEntry.m_pTarget2.GetName(sEntry.m_szTarget2Name, sizeof(sEntry.m_szTarget2Name));
+			sEntry.m_bSuicide = false;
+		}
+		else
+		{
+			if (pAttacker == pVictim)
+			{
+				// Check if entity suicides is enabled.
+				if (g_sDnData.m_piNoticeSuicides.IntValue < 2)
+					return false;
+				
+				sEntry.m_pTarget2 = sEntry.m_pTarget1;
+				sEntry.m_szTarget2Name = sEntry.m_szTarget1Name;
+				sEntry.m_bSuicide = true;
+			}
+			else
+			{
+				// Check if entities killing entities is enabled.
+				if (g_sDnData.m_piNoticeEntityKills.IntValue != 2)
+					return false;
+
+				// The victim is a entity so a bot will have to be used.
+				sEntry.m_pTarget2 = DnManager.GetBot2();
+				if (g_sDnData.m_piNoticeLevel.IntValue == 2 && !sEntry.m_pTarget2.IsValid())
+					return false;
+				
+				DnManager.GetDisplayNameForEvent(pVictim, sEntry.m_szTarget2Name, sizeof(sEntry.m_szTarget2Name));
+				sEntry.m_bSuicide = false;
+			}
+		}
+		return true;
+	}
+	// Hook callback that detects and handles player and entity kills.
+	//
+	public static void EntityKilled(Event hEvent)
+	{
+		if (!CoopManager.IsCoopModeEnabled())
+    	    return;
+
+		CBaseEntity pAttacker = CBaseEntity(hEvent.GetInt("entindex_attacker"));
+		CBaseEntity pVictim = CBaseEntity(hEvent.GetInt("entindex_killed"));
+		CBaseEntity pInflictor = CBaseEntity(hEvent.GetInt("entindex_inflictor"));
+
+		if (!pAttacker.IsValid() || !pVictim.IsValid() || !DnManager.IsEntityAllowed(pVictim))
+			return;
+		
+		DnEntry sEntry;
+		if (pAttacker.IsClassPlayer())
+		{
+			if (!DnManager.WriteEntryPlayerKill(sEntry, view_as<CBasePlayer>(pAttacker), pVictim))
+				return;
+		}
+		else
+		{
+			if (!DnManager.WriteEntryEntityKills(sEntry, pAttacker, pVictim))
+				return;
+		}
+		DnManager.GetWeaponNameForEvent(pInflictor, sEntry.m_szWeapon, sizeof(sEntry.m_szWeapon));
+		DnManager.PushDeathNotice(sEntry);
+	}
+	// Pushes a death notice entry onto the display.
+	//
+	public static void PushDeathNotice(DnEntry sEntry)
+	{
+		if (g_sDnData.m_piNoticeLevel.IntValue == 2)
+		{
+			// Delay is used to allow time for the game to update the player name.
+			// If the death notification displays before the player name is
+			// updated, then the old name will be displayed which is unintended behavior.
+			float flDelay = 0.35; 
+
+			// Pushes a entry a new queue if a queue does not exist.
+			// If a queue does exist, pushes a entry to the queue. 
+			g_sDnData.m_sList.PushArray(sEntry, sizeof(sEntry));
+			if (g_sDnData.m_hTimer == INVALID_HANDLE)
+			{
+				g_sDnData.m_hTimer = CreateTimer(flDelay, DeathNotice_Timer, _, TIMER_REPEAT);
```

Comment:
Not sure if mapchange handling is needed for the timer here

Reply:
Added removal of timer data in for good practice. I don't think anyone will be joining that fast either way lol